src/streamlink/plugins/openrectv.py

```diff
@@ -133,7 +137,7 @@ def _get_streams(self):
                 m3u8_file = subs_data["data"]["items"][0]["media"]["url"]
             # streaming
             elif mdata["onair_status"] == 1:
-                m3u8_file = mdata["media"]["url_ull"]
+                m3u8_file = mdata["media"]["url_ull"] or mdata["subs_trial_media"]["url_ull"]
```

Comment:
This will raise a `KeyError` if the `media` or `subs_trial_media` items do not exist on the `mdata` dict when trying to access their `url_ull` item. Both `media` and `subs_trial_media` are set to optional in the validation schema, so it's not guaranteed that `mdata["media"]` and `mdata["subs_trial_media"]` exist. `media` is the main problem here if you're using `subs_trial_media` as a fallback, unless `media` is always included in the API response. That makes flagging it as optional questionable though.

The whole plugin is in a rather bad shape and could need a proper cleanup/rewrite. For example, it queries the site's API every time for each metadata lookup (stream title and author) in addition to the regular stream URL lookup. The validation schemas are outdated and their return values are being read everywhere in an insecure way, as mentioned above. In addition to that, it uses outdated HTTPSession methods and the overall code style of how error messages are logged is not ideal either. This could all be done in a much simpler/better way.

That's all unrelated to this PR though, so please ignore what I just said.

What needs to be fixed however is how the `url_ull` items are read from the `mdata["media"]` and `mdata["subs_trial_media"]` dictionaries from the validation schema return value.

Reply:
@bastimeyer : both `media` and `subs_trial_media` are indeed always includes in the API call AFAIK.
If no media available (as in this case), it would just be `null` for all the properties (URLs).

Anything needs to be done in this case? Simply mark it as non-optional?

Reply:
> If no media available (as in this case), it would just be `null` for all the properties (URLs).

It should be fine keeping it as optional then. This can be fixed in a proper plugin cleanup at some point later if someone decides to do it.

Reply:
You'll have to fix the trailing whitespace in the validation schema though.

Reply:
Thanks, that's was a rookie mistake.